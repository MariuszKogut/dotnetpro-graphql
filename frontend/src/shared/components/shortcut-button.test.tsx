import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import ShortCutButton from './shortcut-button'

describe('ShortCutButton', () => {
  it('renders without errors', () => {
    const { container } = render(
      <ShortCutButton keyboardShortcut="b">Button</ShortCutButton>,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with bootstrap specific options', () => {
    const { container } = render(
      <ShortCutButton keyboardShortcut="b" variant="secondary">
        Button
      </ShortCutButton>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should call onClick when button was clicked', () => {
    const clickHandler = jest.fn()

    render(
      <ShortCutButton keyboardShortcut="b" onClick={clickHandler}>
        Button
      </ShortCutButton>,
    )

    fireEvent.click(screen.getByText('Button'))

    expect(clickHandler).toBeCalledTimes(1)
  })

  it('should not call onClick when button was not clicked', () => {
    const clickHandler = jest.fn()

    render(
      <ShortCutButton keyboardShortcut="b" onClick={clickHandler}>
        Button
      </ShortCutButton>,
    )

    expect(clickHandler).toBeCalledTimes(0)
  })
})
