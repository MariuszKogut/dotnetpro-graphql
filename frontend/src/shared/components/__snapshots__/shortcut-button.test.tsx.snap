// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortCutButton renders with bootstrap specific options 1`] = `
<div>
  <button
    class="btn btn-secondary"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`ShortCutButton renders without errors 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  >
    Button
  </button>
</div>
`;
